builder/googlecompute/step_create_instance.go

```diff
@@ -98,7 +117,7 @@ func (s *StepCreateInstance) Run(ctx context.Context, state multistep.StateBag)
 
 	var errCh <-chan error
 	var metadata map[string]string
-	metadata, err = c.createInstanceMetadata(sourceImage, string(c.Comm.SSHPublicKey))
+	metadata, err = c.createInstanceMetadata(sourceImage, string(c.Comm.SSHPublicKey), state)
```

Comment:
Hm, here instead of passing the state, I think returning on an eventual error would be better as we would like in the future to diminish the usage of the statebag.